### DISCUSSION TITLE
Eslint rule "react-query/exhaustive-deps"

### DISCUSSION THREAD

**Discussion by Dema** [TanStack/query#1490] [Ideas]
Now as we are using inline functions for `useQuery`, I think there's great need in eslint plugin similar to `"react-hooks/exhaustive-deps"`. Because it is very easy to forgot to include variables used in useQuery's QueryFunction parameter in the key array. This is why actually I prefer the old syntax.

**Comment by vltansky** [comment]
Can be closed now :)
https://tanstack.com/query/v4/docs/react/eslint/eslint-plugin-query
